sound/soc/sof/control.c

```diff
@@ -300,6 +300,10 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 	const struct snd_ctl_tlv __user *tlvd =
 		(const struct snd_ctl_tlv __user *)binary_data;
 
+	/* make sure we have at least a header */
+	if (size < sizeof(const struct snd_ctl_tlv))
+		return -EINVAL;
```

Comment:
-ENOSPC

Reply:
this one I am not sure: this is for put, it's not a matter of space to receive data, it's that the header is too small to be further transmitted to the DSP. I left it unmodified.

Reply:
I am not sure if we should return for this kind of **user space** no space, user should make sure it provide enough memory, if it is inconsistent with the 'size' provided, it is actually 'invalid' arguments?

Reply:
Fair enough. snd_ctl_elem_tlv_write() in alsa-lib doesn't take size as an argument, so you'd hit this error only in case of modified/misconfigured alsa-lib and the application using alsa-lib would not benefit from the error message. So let's stick with -EINVAL here.